pkg/controller/pod_iptables_fip.go

```diff
@@ -0,0 +1,353 @@
+			return err
+		}
+		klog.V(3).Infof("handle add pod annotated iptables fip %s", fipName)
+		if err := c.createOrUpdateCrdFip(fipName, fipName, pod.Annotations[util.IpAddressAnnotation]); err != nil {
+			klog.Errorf("failed to create fip %s: %v", fipName, err)
+			return err
+		}
+	}
+	newPod := pod.DeepCopy()
+	newPod.Annotations[util.FipNameAnnotation] = fipName
+	patch, err := util.GenerateStrategicMergePatchPayload(cachedPod, newPod)
+	if err != nil {
+		return err
+	}
+	if _, err := c.config.KubeClient.CoreV1().Pods(namespace).Patch(context.Background(), name,
+		types.StrategicMergePatchType, patch, metav1.PatchOptions{}, ""); err != nil {
+		if k8serrors.IsNotFound(err) {
+			return nil
+		}
+		klog.Errorf("patch pod %s/%s annotation eip failed: %v", name, namespace, err)
+		return err
+	}
+	return nil
+}
+
+func (c *Controller) handleDeletePodAnnotatedIptablesFip(pod *v1.Pod) error {
+	var err error
+	var keepFipCR bool
+	klog.V(3).Infof("handle delete annotated iptables fip for pod %s/%s", pod.Namespace, pod.Name)
+	if ok, sts := isStatefulSetPod(pod); ok {
+		delete, err := appendCheckPodToDel(c, pod, sts, "StatefulSet")
+		keepFipCR = !isStatefulSetPodToDel(c.config.KubeClient, pod, sts) && !delete && err == nil
+	}
+	if !keepFipCR {
+		fipName := PodNameToEipName(pod.Name, pod.Namespace)
+		if v, ok := pod.Annotations[util.EipNameAnnotation]; ok {
+			v = strings.Trim(v, " ")
+			if v != "" {
+				fipName = v
+			}
+		}
+		klog.V(3).Infof("delete pod annotated iptables fip cr %s", fipName)
+		if err = c.config.KubeOvnClient.KubeovnV1().IptablesFIPRules().Delete(context.Background(), fipName, metav1.DeleteOptions{}); err != nil {
+			if !k8serrors.IsNotFound(err) {
+				klog.Errorf("failed to delete iptables fip %s: %v", fipName, err)
+				return err
+			}
+		}
+	}
+	if err = c.handleDelPodAnnotatedIptablesFipFinalizer(pod); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (c *Controller) handleAddPodAnnotatedIptablesFipFinalizer(pod *v1.Pod) error {
+	if pod.DeletionTimestamp.IsZero() {
+		if util.ContainsString(pod.Finalizers, util.FipFinalizer) {
+			return nil
+		}
+		oriPod := pod.DeepCopy()
+		pod.Finalizers = append(pod.Finalizers, util.FipFinalizer)
+		patch, err := util.GenerateStrategicMergePatchPayload(oriPod, pod)
+		if err != nil {
+			return err
+		}
+		if _, err := c.config.KubeClient.CoreV1().Pods(pod.Namespace).Patch(context.Background(), pod.Name,
+			types.StrategicMergePatchType, patch, metav1.PatchOptions{}, ""); err != nil {
+			if k8serrors.IsNotFound(err) {
+				return nil
+			}
+			klog.Errorf("failed to add iptables eip finalizer for pod %s: %v", pod.Name, err)
+			return err
+		}
+		// wait local cache ready
+		time.Sleep(2 * time.Second)
+	}
+	return nil
+}
+
+func (c *Controller) handleDelPodAnnotatedIptablesFipFinalizer(pod *v1.Pod) error {
+	if len(pod.Finalizers) == 0 {
+		return nil
+	}
+	oriPod := pod.DeepCopy()
+	pod.Finalizers = util.RemoveString(pod.Finalizers, util.FipFinalizer)
```

Comment:
should write the oriPod to avoid write to the cache

Reply:
fixed